comet-java-client/src/main/java/ml/comet/experiment/ApiExperiment.java

```diff
@@ -0,0 +1,137 @@
+        this.experimentKey = experimentKey;
+        if (logger != null) {
+            this.logger = logger;
+        }
+        this.connection = ConnectionInitializer.initConnection(this.config, this.apiKey, this.logger);
+    }
+
+    public ApiExperiment(String experimentKey) {
+        this.experimentKey = experimentKey;
+        this.config = ConfigUtils.getDefaultConfigFromClassPath();
+        this.apiKey = config.getString(COMET_API_KEY);
+        this.connection = ConnectionInitializer.initConnection(this.config, this.apiKey, this.logger);
+    }
+
+    public static ApiExperiment.ApiExperimentBuilderImpl builder(String experimentKey) {
+        return new ApiExperiment.ApiExperimentBuilderImpl(experimentKey);
+    }
+
+    public static class ApiExperimentBuilderImpl implements ApiExperimentBuilder {
+        private final String experimentKey;
+        private String apiKey;
+        private Logger logger;
+        private Config config;
+
+        private ApiExperimentBuilderImpl(String experimentKey) {
+            this.config = ConfigUtils.getDefaultConfigFromClassPath();
+            this.apiKey = config.getString(COMET_API_KEY);
+            this.experimentKey = experimentKey;
+        }
+
+        @Override
+        public ApiExperiment.ApiExperimentBuilderImpl withApiKey(String apiKey) {
+            this.apiKey = apiKey;
+            return this;
+        }
+
+        @Override
+        public ApiExperiment.ApiExperimentBuilderImpl withLogger(Logger logger) {
+            this.logger = logger;
+            return this;
+        }
+
+        @Override
+        public ApiExperiment.ApiExperimentBuilderImpl withConfig(File overrideConfig) {
+            this.config = ConfigUtils.getConfigFromFile(overrideConfig)
+                    .withFallback(this.config)
+                    .resolve();
+            return this;
+        }
+
+        @Override
+        public ApiExperiment build() {
+            return new ApiExperiment(apiKey, experimentKey, logger, config);
+        }
+    }
+
+    @Override
+    protected Connection getConnection() {
+        return this.connection;
+    }
+
+    @Override
+    protected Logger getLogger() {
+        return this.logger;
+    }
+
+    @Override
+    public String getContext() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getWorkspaceName() {
+        return getMetadata().getWorkspaceName();
+    }
+
+    @Override
+    public String getProjectName() {
+        return getMetadata().getProjectName();
+    }
+
+    @Override
+    public String getExperimentName() {
+        return getMetadata().getExperimentName();
+    }
+
+    @Override
+    public String getExperimentKey() {
+        return this.experimentKey;
+    }
+
+    @Override
+    public Optional<String> getExperimentLink() {
+        if (StringUtils.isEmpty(experimentKey)) {
+            return Optional.empty();
+        }
+        String link = config.getString(Constants.BASE_URL_PLACEHOLDER)
```

Comment:
for url construct please use javax.ws.rs.core.UriBuilder

Reply:
Migrated http client to a more up-to-date implementation: OkHttp

Reply:
Updated to construct url through okhttp tools